token/transfer-hook-example/src/state.rs

```diff
@@ -13,3 +16,63 @@ pub fn example_data(account_metas: &[AccountMeta]) -> Result<Vec<u8>, ProgramErr
     ExtraAccountMetas::init_with_account_metas::<ExecuteInstruction>(&mut data, account_metas)?;
     Ok(data)
 }
+
+/// Create the validation data (the extra required accounts configs)
+///
+/// Note: since the PDA is not known at the time of intialization
+/// of the extra required accounts config, we only need to make sure
+/// the account metas were provided in the instruction.
+pub fn create_validation_data(
+    remaining_account_infos: &[AccountInfo],
+    mint_authority: &Pubkey,
+) -> Result<[PodAccountMeta; 4], ProgramError> {
+    if !(remaining_account_infos.len() == 3 // 3 metas and one PDA
+        && remaining_account_infos[0].key == &sysvar::instructions::id()
+        && remaining_account_infos[1].key == mint_authority)
+    {
+        return Err(ProgramError::NotEnoughAccountKeys);
+    }
+    Ok([
+        PodAccountMeta::from(&remaining_account_infos[0]),
+        PodAccountMeta::from(&remaining_account_infos[1]),
+        // Third required account: PDA
+        PodAccountMeta::new_with_seeds(
+            &[
+                Seed::Literal {
+                    bytes: b"transfer-hook-example".to_vec(),
+                },
+                Seed::AccountKey { index: 0 }, // Source account's key
+            ],
+            false,
+            false,
+        )?,
+        PodAccountMeta::from(&remaining_account_infos[2]),
+    ])
+}
```

Comment:
You mean update the transfer hook interface itself? I didn't want to modify it without seeing what you thought of this, but yeah that does tend to make more sense. I was thinking of doing as you said: passing the list of metas/configs as instruction data.

Would this be an issue for people since `PodAccountMeta` does not implement borsh?

I think it's probably unnecessary to check if any of those accounts are signers when initializing the extra metas. After all, there's many use cases where some account is set as an authority but doesn't have to provide a signature until it's needed for some action (metadata update authority)

Reply:
> I was thinking of doing as you said: passing the list of metas/configs as instruction data.

Yeah let's do it! It seems much better to just update the interface.

> Would this be an issue for people since PodAccountMeta does not implement borsh?

I don't think borsh makes any difference. We just need to expose a function like `create_initialize_instruction` which takes `&[PodAccountMeta]` and serialize it however we want, without borsh, and instead as a `PodSlice`.

> I think it's probably unnecessary to check if any of those accounts are signers when initializing the extra metas. After all, there's many use cases where some account is set as an authority but doesn't have to provide a signature until it's needed for some action (metadata update authority)

Great, sounds like we're in agreement!